### PR TITLE
An attemp to update to 1.20.6

### PR DESCRIPTION
**Pull request by EchoEllet**
I sent a [PR](https://github.com/ZsoltMolnarrr/BetterCombat/pull/332) for updating to 1.20.4, and it seems to be working, but this require more work because of more breaking changes

The goal of this PR is to update the mod so it works with 1.20.5 since there are some breaking changes
I will try to update it with care, so when I require java 21 as minimum because of 1.20.5, it's also updated in `README.md`, in GitHub CI, and also in all the other places

if there is something I'm not sure about or something doesn't look right (**Unreachable code** that is already before this update for example) I will always leave a `TODO` so we can look to them later

For now forge doesn't support 1.20.5/6 so to not cause build failure I temporarily disable it, also player animator  dependency mod (I might send a PR right after this) has not updated to 1.20.5 yet

This PR might be closed and never finished since I'm not familiar with the Minecraft Apis and also not with this repository. Player animator dependency has not been updated to 1.20.5/6 yet

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 <div align="center">
 
-  <a href="">![Java 17](https://img.shields.io/badge/Java%2017-ee9258?logo=coffeescript&logoColor=ffffff&labelColor=606060&style=flat-square)</a>
+  <a href="">![Java 21](https://img.shields.io/badge/Java%2021-ee9258?logo=coffeescript&logoColor=ffffff&labelColor=606060&style=flat-square)</a>
```
Comment (ZsoltMolnarrr) [inline README.md:5]: Is the Java version bump necessary?

Comment (EchoEllet) [reply README.md:5]: > Is the Java version bump necessary?

1.20.5 and newer versions require Java 21

Read this for more info: https://www.minecraft.net/en-us/article/minecraft-java-edition-1-20-5

And also this for most of the breaking changes:
https://fabricmc.net/2024/04/19/1205.html

**Comment by ArduFish123** [comment]
Updating this mod is probably impossible until playerAnimator is updated. Maybe there is another mod that does the same thing and gets updated faster?

**Comment by ArduFish** [comment]
1.21 is already released... I think this mod should be updated to 1.21 first.

**Comment by EchoEllet** [comment]
Some dependencies haven't been updated to Minecraft 1.20.5 yet.

[Minecraft 1.21](https://fabricmc.net/2024/05/31/121.html) arrived about a month and a half later, likely introducing new breaking changes.

Minecraft API is not meant to develop new functionalities by developers outside the studio.

Game development and modding are not on my list of interests and probably will never be. Closing this PR as unplanned.

Though if you have any questions, feel free to ask.

**Comment by ZsoltMolnarrr** [comment]
To clarify, Better Combat is not left behind.
It is awaiting PlayerAnimator 1.21 support, and than as soon as I have the free time, I will update it.

Please note that all of these projects are developed on someone’s free time, hence, no timely expectation shall be upheld against any participant :)

**Comment by EchoEllet** [comment]
Looking forward to what comes next. Good Luck.

**Comment by ArduFish123** [comment]
https://github.com/KosmX/minecraftPlayerAnimator/pull/91

**Pull request closed without merging**